### PR TITLE
Python3

### PR DESCRIPTION
**Pull request by jeffmurphy**
Merged: edmunds branch (python3-fixes and nogil) and cisco fork (python3 branch) and then updated, addressed issues like lack of sigint handing, tested against py2 and py3.

### CODE REVIEW

**Comment by alesage** [comment]
Seeing a new warning, any comments @jeffmurphy ?
```
`sig_manager.c:8:12: warning: ‘pynmsg_raise_signal’ used but never defined
 static int pynmsg_raise_signal(int sig);
            ^~~~~~~~~~~~~~~~~~~
```

**Comment by jeffmurphy** [comment]
maybe remove the attribute ... i dont recall why i tagged it with that attribute

https://github.com/farsightsec/pynmsg/blob/python3/sig_manager.c#L7

**Comment by alesage** [comment]
No dice :( , any further thoughts welcome. . . .

**Comment by jeffmurphy** [comment]
Try

```
diff --git a/nmsg_io.pyx b/nmsg_io.pyx
index 71a45b0..382add4 100644
--- a/nmsg_io.pyx
+++ b/nmsg_io.pyx
@@ -28,7 +28,7 @@ def python_check_interrupt(sig, frame):
     global raise_kbint
     raise_kbint = 1
 
-cdef int pynmsg_raise_signal "pynmsg_raise_signal"(int sig) except 0 with gil:
+cdef public int pynmsg_raise_signal "pynmsg_raise_signal"(int sig) except 0 with gil:
     if PyErr_Occurred():
         return 0
     elif sig == SIGINT:
diff --git a/sig_manager.c b/sig_manager.c
index 927dffd..d184e15 100644
--- a/sig_manager.c
+++ b/sig_manager.c
@@ -4,8 +4,7 @@
 #include <string.h>
 #include <Python.h>
 
-__attribute__((unused))
-static int pynmsg_raise_signal(int sig);
+extern int pynmsg_raise_signal(int sig);
 
 static void pynmsg_sighandler(int sig)
 {

```

File: nmsg_input.pyx [hunk 1]
```diff
@@ -13,6 +13,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import threading
+
+def _cstar2str(x):
```
Comment (hstern) [inline nmsg_input.pyx:18]: This looks like a misspelling

**Review by hstern — CHANGES_REQUESTED** [review]
Builds but does not load symbols on macos 10.15.7. Not a blocker for merging.

**Comment by alesage** [comment]
Closing this as it's served by the recent merge of Python3 material to master.

**Pull request closed without merging**